lib/mongodb.js

```diff
@@ -1994,6 +1994,20 @@ function isObjectIDProperty(modelCtor, propDef, value, options) {
   }
 }
 
+/**
+ * Removes extra dollar sign '$' for operators
+ *
+ * @param {*} spec the operator for Where filter
+ */
+function removingDollarSign(spec) {
+  while (spec.charAt(0) === '$') {
+    spec = spec.substr(1);
+  }
+  return spec;
```

Comment:
The function can be simplified as follows:

```js
function trimLeadingDollarSigns(spec) {
  return spec.replace(/^(\$)+/, '');
}
```

Reply:
Indeed! Applied.